umm-spec-lib/src/cmr/umm_spec/umm_g/data_granule.clj

```diff
@@ -18,18 +18,35 @@
          :producer-gran-id (->> Identifiers
                                 (some #(when (= "ProducerGranuleId" (:IdentifierType %)) %))
                                 :Identifier)
+         :crid-ids (->> Identifiers
+                        (filter #(= "CRID" (:IdentifierType %)))
+                        (map :Identifier)
+                        seq) 
+         :feature-ids (->> Identifiers
+                        (filter #(= "FeatureId" (:IdentifierType %)))    
+                        (map :Identifier)
+                        seq)
          :production-date-time ProductionDateTime
          :size (get (first ArchiveAndDistributionInformation) :Size)}))))
 
 (defn DataGranule->umm-g-data-granule
   "Returns the UMM-G DataGranule from the given umm-lib granule model DataGranule."
   [data-granule]
   (when data-granule
-    (let [{:keys [producer-gran-id day-night production-date-time size]} data-granule]
+    (let [{:keys [producer-gran-id crid-ids feature-ids
+                  day-night production-date-time size]} data-granule]
       {:DayNightFlag (string/capitalize day-night)
-       :Identifiers (when producer-gran-id
-                      [{:Identifier producer-gran-id
-                        :IdentifierType "ProducerGranuleId"}])
+       :Identifiers (let [producer-gran-id (when producer-gran-id
+                                             [{:Identifier producer-gran-id
+                                               :IdentifierType "ProducerGranuleId"}])
+                          crid-ids (when (seq crid-ids)
```

Comment:
No need for the call to `when`.

Reply:
Right, since I did check on (seq ) in the end... Done.